rock/python.rb

```diff
@@ -0,0 +1,66 @@
+require 'open3'
+
+module Rock
+    # Identify the available python version, per default use the set values in
+    # Autoproj configuration for 'python_executable' and 'python_version',
+    # otherwise checks Autobuild.programs['python'] setting
+    # @param[String] custom_python_bin Path to a custom python version
+    # @returns[[String,String]] path to the python binary and version, throws if
+    #   python cannot be found
+    def self.find_python_bin(custom_python_bin = nil)
```

Comment:
I'm trying to get rid of globals access as much as possible. A better pattern would be to pass a `ws` keyword argument (an Autoproj::Workspace object, defaulting to Autoproj.workspace) which is then used to resolve environment and configuration within the function.

This will also make the tests cleaner since they won't have to access the global stuff.